.github/workflows/ci.yml

```diff
@@ -0,0 +1,35 @@
+name: CI
+
+# On every pull request, but only on push to master
+on:
+  push:
+    branches:
+    - master
+    tags:
+    - '*'
+  pull_request:
+
+jobs:
+  tests:
+    runs-on: ubuntu-latest
+    container: osgeo/gdal:ubuntu-full-${{ matrix.gdal-version }}
+    strategy:
+      matrix:
+        python-version: [3.7, 3.8, 3.9]
+        gdal-version: [3.4.0, 3.3.3, 3.2.3, 3.1.3, 3.0.4]
+
```

Comment:
Specifying [fail-fast](https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#jobsjob_idstrategyfail-fast) as false could help detecting errors, since all jobs will run until they either fail or complete.
```suggestion
      fail-fast: false

```